### ISSUE TITLE
New HouseCoupon Discount Type

### ISSUE THREAD

**Issue by joelbrock** [CORE-POS/IS4C#524]
```php
            case "%E": // better percent discount applies to specified department only
                Database::getsubtotals();
                $coupon_discount = (int)($infoW['discountValue']*100);
                if ($coupon_discount <= CoreLocal::get('percentDiscount')) {
                    // customer's discount is better than coupon discount; skip
                    // applying coupon
                    $value = 0;
                } else {
                    // coupon discount is better than customer's discount
                    // apply coupon & zero out customer's discount
					
	                $valQ = "select sum(total) from localtemptrans as l 
						left join " . CoreLocal::get('pDatabase') . $transDB->sep() . "houseCouponItems
	                    as h on l.department = h.upc
	                    where h.coupID = " . $coupID . "
	                    and h.type in ('BOTH', 'DISCOUNT')";
	                $valR = $transDB->query($valQ);
	                $row = $transDB->fetch_row($valR);
	                $value = $row[0] * $infoW["discountValue"];					
                }
                break;
```

Also for the Fannie side:  https://github.com/CORESupport/Rutland/commit/e5601ce5e7ac3589d413c274bb0c9620b330944d

**Comment by gohanman** [comment]
I'm not sure that does what's intended. If the customer's discount is 5% and the coupon discount is 10%, they'd get 10% off from the coupon line and an additional 5% off later in the upc=DISCOUNT line.

**Comment by joelbrock** [comment]
it was working as desired in testing.  and i had them test it out too.
I'll have to test more thoroughly to see what you are describing.

**Comment by gohanman** [comment]
![stacked discount](https://cloud.githubusercontent.com/assets/426966/8532634/36df1ebc-23f5-11e5-947f-e9ee543dbaeb.png)

The customer's 5% discount applies to the item immediately on member entry. The 10% coupon discount also applies because 10 > 5.

**Comment by gohanman** [comment]
This seems closer to right: 85bec1ebc3d3a06759ac6d8e73c8fecd7ef77acd

The comment in the original seems questionable "apply coupon & zero out customer's discount". Zeroing out the customer's discount would prevent them from them from receiving a discount on items in other departments (and the code in the first post doesn't actually zero out the customer's discount).

My version flips the coupon-eligible items to non-discountable so the customer's discount does not apply to those items but they still get their discount on any other discountable items.

**Comment by gohanman** [comment]
Included in 1.7
